Lib/test/test_patma.py

```diff
@@ -3332,6 +3419,29 @@ def test_class_pattern_not_type(self):
                     w = 0
         self.assertIsNone(w)
 
+    def test_class_or_union_not_specialform(self):
+        from typing import Literal
+        name = type(Literal).__name__
+        msg = rf"called match pattern must be a class or types.UnionType \(got {name}\)"
+        w = None
+        with self.assertRaisesRegex(TypeError, msg):
+            match 1:
+                case Literal():
+                    w = 0
+        self.assertIsNone(w)
+
+    def test_patma_legacy_union_type(self):
```

Comment:
Please, also test `list[str]` and `List[str]`

Reply:
`list[str]` and `List[str]` are already individually not allowed, but it seems a test for this is missing. Also the error message deviates from isinstance:

```python
>>> t = list[str] 
>>> isinstance(["a"], t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    isinstance(["a"], t)
    ~~~~~~~~~~^^^^^^^^^^
TypeError: isinstance() argument 2 cannot be a parameterized generic
>>> match ["a"]: 
...     case t():
...         pass
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
    case t():
         ~^^
TypeError: called match pattern must be a class or types.UnionType (got types.GenericAlias)
```

Reply:
I added:

- test_generic_type
- test_legacy_generic_type
- test_generic_union_type

the first two feel a bit out of scope of this PR/Issue, anyway...

Reply:
sorry, I've meant `something | list[str]` and `List[str] | something` and etc. like you correctly noted in https://github.com/python/cpython/pull/118644#discussion_r1592516254 :)